src/CountryPicker.js

```diff
@@ -0,0 +1,257 @@
+        )
+        .sort((a, b) => (a.callingCode > b.callingCode ? 1 : -1)),
+    });
+  };
+
+  closePicker = () => {
+    this.setState({
+      showPicker: false,
+    });
+  };
+
+  keyExtractor = (item: CountryCode) => {
+    return item.isoCountryCode;
+  };
+
+  getItemLayout = (data?: CountryCode[] | null, index: number) => {
+    return {
+      length: ITEM_HEIGHT,
+      offset: ITEM_HEIGHT * index,
+      index,
+    };
+  };
+
+  renderItem = ({ item }: ListRenderItemInfo<CountryCode>) => {
+    return (
+      <TouchableOpacity
+        key={item.isoCountryCode}
+        onPress={this.onPressCountry(item)}
+        style={[
+          defaultStyles.item,
+          item.callingCode === this.state.selectedValue
+            ? defaultStyles.selectedItem
+            : null,
+        ]}
+      >
+        <Text key={item.isoCountryCode}>{`${item.flag} ${item.name} +${
+          item.callingCode
+        }`}</Text>
+      </TouchableOpacity>
+    );
+  };
+
+  render() {
+    const {
+      placeholder,
+      placeholderTextColor,
+      style,
+      textStyle,
+      containerStyle,
+      backButtonText,
+      headerTitle,
+      ...rest
+    } = this.props;
+    const { selectedValue, showPicker, countriesCode } = this.state;
+
+    const selectedValueIndex = countriesCode.findIndex(
+      item => item.callingCode === selectedValue
+    );
+    return (
+      <View style={containerStyle}>
+        <TouchableOpacity
+          onPress={this.openPicker}
+          style={[defaultStyles.style, style]}
+        >
+          <Text
+            onFocus={this.openPicker}
+            style={[
+              defaultStyles.textStyle,
+              textStyle,
+              selectedValue ? null : { color: placeholderTextColor },
+            ]}
+          >
+            {selectedValue ? `+${selectedValue}` : placeholder}
+          </Text>
+          <Image source={dropdownArrowIcon} />
+        </TouchableOpacity>
+        <ExtraText {...rest} />
+        <Modal visible={showPicker} animationType="slide">
+          <SafeAreaView style={defaultStyles.container}>
+            <View style={defaultStyles.header}>
+              <TouchableOpacity
+                style={defaultStyles.backButton}
+                onPress={this.closePicker}
+              >
+                <Image source={backIcon} />
+                <Text style={defaultStyles.backButtonText}>
+                  {backButtonText || "Back"}
+                </Text>
+              </TouchableOpacity>
+              <Text style={defaultStyles.headerTitle}>
+                {headerTitle || "Select Country"}
+              </Text>
+            </View>
+            <View style={defaultStyles.searchbarContainer}>
+              <TextInput
+                placeholder="Search"
+                onChangeText={this.search}
+                style={defaultStyles.searchbar}
+                autoFocus={true}
+              />
+            </View>
+            <FlatList
```

Comment:
This list should be extracted out to a standalone component and we should allow developers use other method to show the list other than `Modal` because I rmb frank said we should push a new page for user to select country

Reply:
I create an `openAlternativeCountryList` prop to allow the user to show the other list in this way.
```ts
<CountryPicker
  {...rest}
  openAlternativeCountryList: (
    countryCodes: Country[],
    onSelectCountry: (country: Country) => void
  ) => {
    this.props.navigation.push("CountryList", {
      onSelectCountry,
    });
  },
/>
```